### ISSUE TITLE
Bug: Warning: File ignored because of a matching ignore pattern. (git hook)

### ISSUE THREAD

**Issue by infacto** [eslint/eslint#17436]
### Environment

Node version: v18.17.0
npm version: v9.6.7
Local ESLint version: v8.24.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19045

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I use husky with lint-stage for git hook on commit to execute eslint:

```json
{
  "lint-staged": {
    "*.{ts,js}": "eslint --fix --max-warnings=0"
  }
}
```

Max warnings zero because I use `only-warn` plugin for IDE. So only commit should fail on issue.

Now I want to commit an ignored file from project setup. But this is not possible because of a useless warning:

```
ESLint found too many warnings
0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
```

### What did you expect to happen?

Why care about ignored files? Just proceed and don't throw a warning.

### What actually happened?

A warning appears because the passed file is marked to ignore.

### Link to Minimal Reproducible Example

Not possible or necessary in this case.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Don't warn for ignored files.
